packages/api-server/api_server/routes/delivery_alerts.py

```diff
@@ -58,16 +60,59 @@ async def query_delivery_alerts(
 
     try:
         delivery_alerts = await ttm.DeliveryAlert.filter(**filters)
-        return [
+        delivery_alerts_pydantic = [
             await ttm.DeliveryAlertPydantic.from_tortoise_orm(a)
             for a in delivery_alerts
         ]
     except FieldError as e:
         raise HTTPException(422, str(e)) from e
 
+    ongoing_delivery_alerts = []
+    # Check for dangling delivery alerts, where none of the fleets are still
+    # performing the task
+    current_task_ids = []
+    try:
+        db_states = await ttm.FleetState.all().values_list("data", flat=True)
+        fleets = [FleetState(**s) for s in db_states]
+        for f in fleets:
+            if f.robots is None:
+                continue
+            for robot in f.robots.values():
+                if robot.task_id is not None and len(robot.task_id) != 0:
+                    current_task_ids.append(robot.task_id)
+    except Exception as e:  # pylint: disable=broad-except
```

Comment:
For this particular scenario, where there's a lot of action happening under-the-hood, especially with casting involved, I would prefer a broad except, rather than extensively writing out every single possible exception.

Reply:
You should still write out the exceptions which you can handle, let uncaught exceptions propagate up to fastapi and it should return with an http error code.

Although even if we were to manage the states ourselves (which we should not), this should be done when we ingest the task states, querying should not cause this kind of side effects.